File: aplicação/frontend/ea-project/src/models/output/GetAllTrialsOutput.js
import BaseOutput from './BaseOutput';

export default class GetAllTrialsOutput extends BaseOutput {
    constructor(getalltrialList = [], feedbackMessages = []) {
        super(feedbackMessages);
        this.getalltrialList = getalltrialList.map(
            (getalltrial) => new GetAllTrialsOutput.GetAllTrialProperties(
                getalltrial.id,
                getalltrial.name,
                getalltrial.startDate,
                getalltrial.distance,
                getalltrial.distanceUnit,
                getalltrial.isActive,
                getalltrial.creationDate
            )
        );
    }

    static GetAllTrialProperties = class {
        constructor(id, name, startDate, distance, distanceUnit, isActive, creationDate) {
            this.id = id;
            this.name = name;
            this.startDate = startDate;
            this.distance = distance;
            this.distanceUnit = distanceUnit;
            this.isActive = isActive;
            this.creationDate = creationDate;
        }
    };
}

File: app/backend/src/main/java/eaproject/output/GetAllTrialsOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.ArrayList;

public class GetAllTrialsOutput extends BaseOutput implements Serializable {

    @JsonProperty
    private ArrayList<TrialProperties> trials;

    public ArrayList<TrialProperties> getTrials() {
        return trials;
    }

    public void setTrials(ArrayList<TrialProperties> trials) {
        this.trials = trials;
    }

    public static class TrialProperties implements Serializable {

        @JsonProperty
        private Integer id;

        @JsonProperty
        private String name;

        @JsonProperty
        private Timestamp startDate;

        @JsonProperty
        private Double distance;

        @JsonProperty
        private String distanceUnit;

        @JsonProperty
        private boolean isActive;

        @JsonProperty
        private double lon;

        @JsonProperty
        private double lat;

        @JsonProperty
        private String location;

        @JsonProperty
        private String modality;

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public Timestamp getStartDate() {
            return startDate;
        }

        public void setStartDate(Timestamp startDate) {
            this.startDate = startDate;
        }

        public Double getDistance() {
            return distance;
        }

        public void setDistance(Double distance) {
            this.distance = distance;
        }

        public String getDistanceUnit() {
            return distanceUnit;
        }

        public void setDistanceUnit(String distanceUnit) {
            this.distanceUnit = distanceUnit;
        }

        public boolean getIsActive() {
            return isActive;
        }

        public void setIsActive(boolean isActive) {
            this.isActive = isActive;
        }

        public double getLon() {
            return lon;
        }

        public void setLon(double lon) {
            this.lon = lon;
        }

        public double getLat() {
            return lat;
        }

        public void setLat(double lat) {
            this.lat = lat;
        }

        public String getLocation() {
            return location;
        }

        public void setLocation(String location) {
            this.location = location;
        }

        public String getModality() {
            return modality;
        }

        public void setModality(String modality) {
            this.modality = modality;
        }
    }
}
